source/js/interview.js

```diff
@@ -0,0 +1,270 @@
+            console.error(err);       
+        }
+
+    }
+
+    // Checks whether the card provided is or isn't already loaded
+    function isCardAlreadyOpened(ID) {
+        if ($("#" + ID).hasClass("active"))
+            return true;
+        else
+            return false;
+    }
+
+    // Function that's responsible for calling the opening and closing function for the modal box with the right parameters
+    function cardTogglePC(sender) {
+        if (sender != null)
+            modal_box.html(sender.innerHTML.replace(/no-display/g, ""));
+        else
+            modal_box.html(sender);
+    }
+
+    // Function that's responsible for opening and closing the student card on a mobile device
+    function cardTogglePhone(sender) {
+        if(sender == null) 
+            throw new Error("[" + cardTogglePhone.name + "] > NullReferenceException: Object reference not set to an instance of an object");
+
+        $("body").toggleClass("no-overflow");
+        $("#" + sender.id).toggleClass("opened");
+        $(`#${sender.id} .card-body .card-text .card-long-desc`).toggleClass("no-display");
+        $(`#${sender.id} .card-body h2`).toggleClass("no-display");
+        $(`#${sender.id} .card-body .card-images`).toggleClass("no-display");
+        closeBtn.toggleClass("no-display");
+        currentCardPhone = sender;
+        iscardOpenPhone = !iscardOpenPhone;
+    }
+
+    /**
+     * @description Checks whether the current document dimensions fulfil the requirements to be considered a desktop viewport
+     * @returns Boolean
+     */
+    function isDesktop() {
+        if (window.innerWidth >= 670 && window.innerHeight >= 300)
+            return true;
+        else
+            return false;
+    }
+
+    // Function that's responsible for generating hash from a string 
+    function hashCode(string) {
+        for(var i = 0, h = 0; i < string.length; i++)
+            h = Math.imul(31, h) + string.charCodeAt(i) | 0;
+    
+        return h;
+    }
+
+    // Function that's responsible for checking the URL for any arguments/parameters and taking actions accordingly
+    function checkUrlForRequest() {
+        var loadcard = null;
+        var url = new URL(window.location.href);
+        if(!url.search || url.search == null)
+            return;
+        var id = url.searchParams.get("id");
+        if(id && id != null) {
+            for(var card of cards) {
+                if(card.id == "modal-box")
+                    break;
+                
+                if(hashCode(card.id) == id)
+                    loadcard = card;
+
+                if(hashCode(card.id) == hashCode(id))
+                    loadcard = card;
+
+                if(hashCode(card.id.toLowerCase()) == hashCode(id.toLowerCase()))
+                    loadcard = card;
+            }
+
+            if(loadcard === null)
+                alert(`"${id}" is invalid!`);
```

Comment:
is this the browser popup? if yes, then please don't :) browser popups are very annoying